test/coreInternal.spec.ts

```diff
@@ -0,0 +1,265 @@
+import * as chai from "chai";
+import * as _ from "lodash";
+
+import * as ABIDecoder from "abi-decoder";
+import { BigNumber } from "bignumber.js";
+import { ether } from "./utils/units";
+
+// Types
+import { Address } from "../types/common.js";
+
+// Contract types
+import { CoreContract } from "../types/generated/core";
+import { SetTokenContract } from "../types/generated/set_token";
+import { SetTokenFactoryContract } from "../types/generated/set_token_factory";
+import { StandardTokenMockContract } from "../types/generated/standard_token_mock";
+import { TransferProxyContract } from "../types/generated/transfer_proxy";
+import { VaultContract } from "../types/generated/vault";
+
+// Artifacts
+const Core = artifacts.require("Core");
+
+// Core wrapper
+import { CoreWrapper } from "./utils/coreWrapper";
+
+// Testing Set up
+import { BigNumberSetup } from "./config/bignumber_setup";
+import ChaiSetup from "./config/chai_setup";
+BigNumberSetup.configure();
+ChaiSetup.configure();
+const { expect } = chai;
+
+import {
+  expectRevertError,
+} from "./utils/tokenAssertions";
+
+import {
+  STANDARD_NATURAL_UNIT,
+} from "./constants/constants";
+
+contract("CoreInternal", (accounts) => {
+  const [
+    ownerAccount,
+    otherAccount,
+  ] = accounts;
+
+  let core: CoreContract;
+  let transferProxy: TransferProxyContract;
+  let vault: VaultContract;
+  let setTokenFactory: SetTokenFactoryContract;
+
+  const coreWrapper = new CoreWrapper(ownerAccount, ownerAccount);
+
+  before(async () => {
+    ABIDecoder.addABI(Core.abi);
+  });
+
+  after(async () => {
+    ABIDecoder.removeABI(Core.abi);
+  });
+
+  beforeEach(async () => {
+    core = await coreWrapper.deployCoreAsync();
+  });
+
+  describe("#setVaultAddress", async () => {
+    beforeEach(async () => {
+      vault = await coreWrapper.deployVaultAsync();
+      await coreWrapper.addAuthorizationAsync(vault, core.address);
+    });
+
+    let caller: Address = ownerAccount;
+
+    async function subject(): Promise<string> {
+      return core.setVaultAddress.sendTransactionAsync(
+        vault.address,
+        { from: caller },
+      );
+    }
+
+    it("sets vault address correctly", async () => {
+      await subject();
+
+      const storedVaultAddress = await core.vaultAddress.callAsync();
+      expect(storedVaultAddress).to.eql(vault.address);
+    });
+
+    describe("when the caller is not the owner of the contract", async () => {
+      beforeEach(async () => {
+        caller = otherAccount;
```

Comment:
the way that this is currently setup, the `caller` will be `otherAccount` after this test is run instead of being reset to `ownerAccount` like it should be?